pilot/pkg/bootstrap/server.go

```diff
@@ -755,12 +761,23 @@ func (s *Server) initConfigRegistry(serviceControllers *aggregate.Controller) {
 }
 
 func (s *Server) initDiscoveryService(args *PilotArgs) error {
-	environment := &model.Environment{
-		Mesh:             s.mesh,
-		IstioConfigStore: s.istioConfigStore,
-		ServiceDiscovery: s.ServiceController,
-		ServiceAccounts:  s.ServiceController,
-		MixerSAN:         s.mixerSAN,
+	var environment *model.Environment
+	if mcpEnabled(args) {
```

Comment:
mcp enabled means it'll also fetch from mcp - I assume the other controlers will still be enabled.